File: set_tmux.py
#!/usr/bin/env python3

import os
import shutil
import sys
import subprocess


def set_tmux(file):

    home = os.path.expanduser('~')
    location_tmux_conf = home + "/." + file
    backup_tmux_conf = home + "/." + file + ".bk"
    plugin_location = home + "/.tmux/plugins/tpm"

    if os.path.exists(location_tmux_conf):
        shutil.copy(location_tmux_conf, backup_tmux_conf)
        shutil.copy(file, location_tmux_conf)
    else:
        shutil.copy(file, location_tmux_conf)

    if not os.path.exists(plugin_location):
        subprocess.check_output(
            f"git clone https://github.com/tmux-plugins/tmp {plugin_location}",
            shell=True)
    else:
        print("tpm location is set")


if __name__ == '__main__':
    if len(sys.argv) == 2:
        file = sys.argv[1]
        set_tmux(file)
    else:
        print("set_tmux file_name")

File: set_tmux.rb
#!/usr/env/env ruby

def set_tmux(file)
  home = ENV['HOME']
  location_tmux_conf = "#{home}/.#{file}"
  backup_tmux_conf = "#{home}/.#{file}.bk"
  tmux_plugins_location = "#{home}/.tmux/plugins/tpm"

  if !File.exist?(location_tmux_conf)
    `cp #{file} #{location_tmux_conf}`
  else
    `cp #{location_tmux_conf} #{backup_tmux_conf}`
    `cp #{file} #{location_tmux_conf}`
  end

  if !File.exist?(tmux_plugins_location)
    `git clone https://github.com/tmux-plugins/tpm #{tmux_plugins_location}`
  else
    puts 'tpm location is set'
  end
end

def main
  if ARGV[0]
    file = ARGV[0]
    set_tmux(file)
  else
    puts 'ruby set_tmux.rb tmux.conf'
  end
end

main if __FILE__ == $PROGRAM_NAME
